Snakemake/rules/merge_samples_vcf.smk

```diff
@@ -4,113 +4,108 @@
 #
 import itertools
 import numpy as np
-configfile: 'test.yaml'
+configfile: 'config.yaml'
 
-def list_full_paths(directory):
-    return [os.path.join(directory, file) for file in os.listdir(directory)]
+dir = config['vcfDir']
+# single file patterns
+(PREFIX, SUFFIX) = glob_wildcards(dir+'/{prefix}_{suffix}.vcf')
+#print(PREFIX)
 
-def dic_files_per_chromosome(direc_lst, noChromosomes):
-    file_dic={}
-    for i in range(1, noChromosomes + 1):
-        file_dic[i] = [name for name in direc_lst if str(i) in name]
-
-    return file_dic
 
 rule all:
     input:
-        expand('Chr{chromosome}.vcf.gz', chromosome=range(1, config['noChromosomes'] + 1))
-
-#workdir: config['vcfDir']
+        expand('{prefix}_{suffix}', zip, prefix=(PREFIX), suffix=SUFFIX)
```

Comment:
What is "zip" here? Do you define it somewhere?

Reply:
Ah, ok! I know about zip - and this is the syntax, you don't have to do zip(prefix, suffix)?